src/identity/mnemonic.ts

```diff
@@ -158,6 +159,52 @@ export class Mnemonic extends MasterPublicKey implements IdentityInterface {
     return pset.toBase64();
   }
 
+  async signPsetV2(psetBase64: string): Promise<string> {
+    if (!this.ecclib) throw new Error('ecclib is missing, cannot sign pset');
+
+    const pset = Pset.fromBase64(psetBase64);
+    const signer = new Signer(pset);
+
+    let i = 0;
+    for (const input of pset.inputs) {
+      const sighashType = input.sighashType;
+      if (sighashType === undefined) {
+        throw new Error(`Missing sighash type for input ${i}`);
+      }
+      const prevout = input.getUtxo();
+      if (!prevout) {
+        throw new Error(`Missing prevout for input ${i}`);
+      }
+
+      const addressGeneration = this.scriptToAddressCache[
+        prevout.script.toString('hex')
+      ];
+
+      if (addressGeneration) {
+        const privateKeyBuffer = this.derivePath(
+          addressGeneration.address.derivationPath!
+        ).privateKey;
+        const signingKeyPair = ECPairFactory(this.ecclib).fromPrivateKey(
+          privateKeyBuffer
+        );
+        const preimage = pset.getInputPreimage(i, sighashType);
```

Comment:
Maybe here we should decode the script type? if is segwit v1 we should branch out, and at least if is a key-path spend try to make the schnorr signature necessary?

Reply:
If not in this PR, at least let's throw an error that is not implemented in the case is a taproot output and/or `input.tapLeafScript` is present, so user knows can only signs segwit v0

Reply:
@altafan @louisinger